scripts/apps/authoring-react/authoring-react.tsx

```diff
@@ -452,22 +452,28 @@ export class AuthoringReact<T extends IBaseRestApiResponse> extends React.PureCo
         return itemWithFieldsApplied;
     }
 
-    handleFieldChange(fieldId: string, data: unknown) {
+    handleFieldChange(fieldId: string, data: unknown, exposed: IExposedFromAuthoring<T>) {
         const {state} = this;
 
         if (state.initialized !== true) {
             throw new Error('can not change field value when authoring is not initialized');
         }
 
-        const {onFieldChange} = this.props;
         const fieldsDataUpdated = state.fieldsDataWithChanges.set(fieldId, data);
+        const resultFromOnFieldChange = this.props.onFieldChange?.(
```

Comment:
Avoid optional chaining for function calls, especially when you expect something returned. Here for example `resultFromOnFieldChange` can evaluate to `undefined` and on next line you access its properties as it were an object.

Reply:
I agree, but in this case I've handled it on the next lines, so it's ok

Reply:
![image](https://github.com/superdesk/superdesk-client-core/assets/2076953/bce7f734-b5d2-4211-a3d2-2af26120eedc)

Doesn't seem handled. If you want to keep the conditional function call at least add the type:
`const resultFromOnFieldChange: undefined | MyType = ...`